tests/test_serialization_options.py

```diff
@@ -0,0 +1,91 @@
+# Copyright 2021 Google LLC.
+# Copyright (c) Microsoft Corporation.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from test_helpers import execute_command
+
+
+@pytest.mark.parametrize("eval, max_object_depth, expected_result", [
+    ("({'foo': {'bar': 'baz'}})", 0, {
+        "type": "object"
+    }),
+    ("({'foo': {'bar': 'baz'}})", 1, {
+        "type": "object",
+        "value": [["foo", {
+            "type": "object",
+        }]],
+    }),
+    ("({'foo': {'bar': 'baz'}})", 2, {
+        "type": "object",
+        "value": [[
+            "foo", {
+                "type": "object",
+                "value": [["bar", {
+                    "type": "string",
+                    "value": "baz"
+                }]]
+            }
+        ]],
+    }),
+    ("({'foo': {'bar': 'baz'}})", None, {
+        "type": "object",
+        "value": [[
+            "foo", {
+                "type": "object",
+                "value": [["bar", {
+                    "type": "string",
+                    "value": "baz"
+                }]]
+            }
+        ]],
+    }),
+])
+@pytest.mark.asyncio
+async def test_serializationOptions(websocket, context_id, eval,
+                                    max_object_depth, expected_result):
+    result = await execute_command(
+        websocket, {
+            "method": "script.evaluate",
+            "params": {
+                "expression": eval,
+                "target": {
+                    "context": context_id
+                },
+                "awaitPromise": False,
+                "serializationOptions": {
+                    "maxObjectDepth": max_object_depth
+                }
+            }
+        })
+    assert result["result"] == expected_result
+
+    result = await execute_command(
+        websocket, {
+            "method": "script.callFunction",
+            "params": {
+                "functionDeclaration": f"()=>{eval}",
+                "this": {
+                    "type": "undefined"
+                },
+                "awaitPromise": False,
+                "target": {
+                    "context": context_id
+                },
+                "serializationOptions": {
+                    "maxObjectDepth": max_object_depth
+                }
+            }
+        })
+    assert result["result"] == expected_result
```

Comment:
From a unit test perspective, it would be cleaner to split `callFunction` into separate tests, IMHO.

There are two ways to do this: You could either introduce a boolean param + add an `if` (to execute either `callFunction` or `evaluate`), or you could split the method into two (`test_serializationOptions_script_evaluate` and `test_serializationOptions_script_call_function).

The former would reduce code duplication, but the latter would be more readable.

I'd suggest to adopt an approach like in the [`pytest` docs](https://docs.pytest.org/en/7.1.x/example/parametrize.html#different-options-for-test-ids), with a `TESTDATA = ` definition at the top:

```
testdata = [
    (datetime(2001, 12, 12), datetime(2001, 12, 11), timedelta(1)),
    (datetime(2001, 12, 11), datetime(2001, 12, 12), timedelta(-1)),
]


@pytest.mark.parametrize("a,b,expected", testdata)
def test_timedistance_v0(a, b, expected):
    diff = a - b
    assert diff == expected


@pytest.mark.parametrize("a,b,expected", testdata, ids=["forward", "backward"])
def test_timedistance_v1(a, b, expected):
    diff = a - b
    assert diff == expected
```

Then we can have the two separate methods for each script calling style and they would be quite readable.

Reply:
good idea. Done